### PR TITLE
Sgt back end

### PR DESCRIPTION
**Pull request by CTMendoza**
sgt-back-end

### CODE REVIEW

File: sgt-back-end/index.js [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+    });
+});
+
+// POST - inserts a new grade into "grades" table and return the created grade.
+// The client should receive and object, not an array
+app.post('/api/grades', (req, res, next) => {
+  // status 400 if client supplies an invalid grade, including a missing name, course or score.
+  if (!req.body || !req.body.name || !req.body.course || !req.body.score) {
+    const errObj = { error: 'missing all or some of name, course, score query parameters' };
+    return res.status(400).json(errObj);
```
Comment (TimelikeClosure) [inline sgt-back-end/index.js]: This will reject a `score` of `0`, which should be valid.

File: sgt-back-end/index.js [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+  } else if (req.body.score < 0 || req.body.score > 100 || isNaN(req.body.score)) { // Or score isnt an integer from 0 to 10
+    const errObj = { error: 'score value is not an integer from 0 to 100 or is NaN' };
+    return res.status(400).json(errObj);
```
Comment (TimelikeClosure) [inline sgt-back-end/index.js]: This won't stop non-integer-numbers in the range (ex: `4.52` is not an integer, but it will pass through).

**Review by TimelikeClosure — CHANGES_REQUESTED** [review]

**Comment by CTMendoza** [comment]
sgt-back-end update

**Review by TimelikeClosure — CHANGES_REQUESTED** [review]
@CTMendoza Please make sure you're validating `score` and `id` values correctly on your `PUT` and `DELETE` routes as well.

**Review by TimelikeClosure — APPROVED** [review]

**Pull request merged**